blog/nse_portfolio_sorts/index.qmd

```diff
@@ -0,0 +1,931 @@
+
+```{r}
+# Create sorting grid
+setup_grid <- expand_grid(sorting_variable = "sv_ag",
+                          drop_smallNYSE_at = c(0, 0.05, 0.2),
+                          include_financials = c(TRUE, FALSE),
+                          include_utilities = c(TRUE, FALSE),
+                          drop_bookequity = c(TRUE, FALSE),
+                          drop_earnings = c(TRUE, FALSE),
+                          drop_stock_age_at = c(0, 2),
+                          drop_price_at = c(0, 1, 5),
+                          sv_lag = c("3m", "6m", "FF"),
+                          formation_time = c("monthly", "FF"),
+                          n_portfolios_main = c(5, 10),
+                          sorting_method = c("single", "dbl_ind", "dbl_dep"),
+                          n_portfolios_secondary = c(2, 5),
+                          exchanges = c("NYSE", "NYSE|NASDAQ|AMEX"),
+                          value_weighted = c(TRUE, FALSE))
+
+# Remove information on double sorting for univariate sorts
+setup_grid <- setup_grid |> 
+  filter(!(sorting_method == "single" & n_portfolios_secondary > 2)) |> 
+  mutate(n_portfolios_secondary = case_when(sorting_method == "single" ~ NA_real_, 
+                                            TRUE ~ n_portfolios_secondary))
+```
+
+## Merge data
+
+One key decision node is the sorting variable lag. However, merging data is an expensive operation, and doing it repeatedly is unnecessary. Hence, we merge the data in the three possible lag configurations and store them as separate tibbles. Thereby, we can later reference the correct table instead of merging the desired output.
+
+First, let us consider the Fama-French (FF) lag. Here, we consider accounting information published in year $t-1$ starting from July of year $t$. That is, we use the accounting information published 6 to 18 months ago. We first match the accounting data to the stock market data before we fill in the missing observations. A few pitfalls exist when using the `fill()`-function. First, one might easily forget to order and group the data. Second, the function does not care how outdated the information becomes. In principle, you can end up with data that is decades old. Therefore, we ensure that these filled data points are not older than 12 months. Finally, notice that this code provides much flexibility. All variables with prefixes *sv_* and *filter_* get filled. So you can easily adapt my code to your needs.
+
+```{r}
+data_FF <- crsp_monthly |> 
+  mutate(sorting_date = month) |> 
+  left_join(compustat |> 
+              mutate(sorting_date = floor_date(datadate, "year") %m+% months(18),
+                     sv_date_COMP_y = sorting_date) |> 
```

Comment:
Column naming nicht klar - wofür steht "sv_date_COMP_y"?

Reply:
Very good point, I clarified it in the text and changed the variable (Commit https://github.com/tidy-finance/website/pull/34/commits/cceb811d6f544e76839e60a59a80902f98c453a9).